deposits.md

```diff
@@ -0,0 +1,225 @@
+
+We define a new [EIP-2718] compatible transaction type with the prefix `0x7E`, and the following
+fields:
+
+- `address to`
+- `address from`
+- `uint256 value`
+- `bytes data`
+- `uint256 gasLimit`
+
+This is a subset of the fields used in [EIP-155], but does not include signature information.
+
+We select `0x7E` because transaction type identifiers are currently allowed to go up to `0x7F`.
+Picking a high identifier minimizes the risk that the identifier will be used by Ethereum in the
+future. We don't pick `0x7F` itself in case it becomes used for a variable-length encoding scheme.
+
+Although in practice we define only one new Transaction Type we can distinguish between two distinct
+transactions which occur in the deposit block, based on their positioning. The first transaction
+MUST be a [L1 attributes deposit][l1-attributes-deposit], followed by a an array of zero-or-more
+[L1 transaction deposits][l1-transaction-deposits] submitted to the Deposit Feed contract by
+accounts on L1.
+
+> **TODO** Specify and link to deposit blocks
+
+### Validation and Authorization of Deposit Transaction Types
+
+[authorization]: #authorization
+
+As noted above, the Deposit Transaction Type does not include a signature for validation. Rather,
+authorization is handled by the [L1 Deposit Feed contract][deposit-feed-contract] and the
+[L2 chain Derivation][derivation] process itself.
+
+If a Deposit Transaction is included which is not derived using the correct derivation algorithm,
+the resulting state transition would be invalid.
+
+### Execution
+
+In order to execute a deposit transaction:
+
+First, the balance of the `from` account MUST be increased by the amount of `value`.
+
+Then, the execution environment for a deposit transaction is initialized based on the transaction's
+values, in exactly the same manner as it would be for an EIP-155 transaction.
+
+Specifically, a new EVM call frame targeting the `to` address is created with values initialized as
+follows:
+
+- `CALLER` and `ORIGIN` set to `from`
+- `context.calldata` set to `data`
+- `context.gas` set to `gasLimit`
+- `context.value` set to `value`
+
+#### Nonce handling
+
+Despite the lack of signature validation, we still increment the nonce of the `from` account when a
+Deposit Transaction is executed. In the context of a deposit-only roll up, this is not necessary
+for transaction ordering or replay prevention, however it maintains consistency with the use of
+nonces during contract creation. It may also simplify integration with downstream tooling (such
+as wallets and block explorers).
+
+## L1 Attributes Deposit
+
+[l1-attributes-deposit]: #l1-attributes-deposit
+
+This is a deposit sent to the [Layer 1 Attributes Predeploy][l1-attributes-predeploy] contract.
+
+This transaction MUST have the following values:
+
+1. `from` is the L1 Attributes Depositor Account `0xdeaddeaddeaddeaddeaddeaddeaddeaddead0001`.
```

Comment:
If `from` and `to` have to be constant values, why are they serialized in the tx? If I understand correctly, `from` in the tx is the L1 `msg.sender` and `to` is the target contract being called from L1 and these hard coded values are set for the EVM context and the original data in the tx is encoded in the calldata

Reply:
This is the L1 attributes deposit transaction, which sends L1 header values to a L2 contract when they can subsequently be read by transsactions in the block. This transacton is always the very first in a L2 block.

Things are hardcoded because the L1 attributes deposit transaction has the same EIP-2718 type as user deposit transactions. It's just generated automatically from L1 blocks instead of being submitted by a user.

The rationale was that it would be easier to process (smaller code diff) to do it this way rather than introducing another separate tx type.

@maurelian maybe this needs to be introduced a bit better? it's a bit free floating right now, which is understandable because these transactions are not processed by the deposit feed contract (it's debatable whether this should be specified here imho) -- let's consider this for a subsequent PR?

Reply:
I modified some of the earlier content to make the distinction between the two uses of this tx type more explicit, and explain the rationale for only creating a single new type.

https://github.com/ethereum-optimism/optimistic-specs/commit/4729f5ec12e92417314d6727d97cd488e3e59292